Tests/SerializationTransformerTests.swift

```diff
@@ -0,0 +1,215 @@
+//
+//  SerializationTransformerTests.swift
+//  Kakapo
+//
+//  Created by Alex Manzella on 27/06/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+import Quick
+import Nimble
+import SwiftyJSON
+@testable import Kakapo
+
+struct UppercaseTransformer<Wrapped: Serializable>: SerializationTransformer {
+    
+    let wrapped: Wrapped
+    
+    func transform(key key: String) -> String {
+        return key.uppercaseString
+    }
+}
+
+struct LowercaseFirstCharacterTransformer<Wrapped: Serializable>: SerializationTransformer {
+    
+    let wrapped: Wrapped
+    
+    func transform(key key: String) -> String {
+        let characters = key.characters
+        let first = String(characters.prefix(1)).lowercaseString
+        let other = String(characters.dropFirst())
+        return first + other
+    }
+}
+
+class SerializationTransformerSpec: QuickSpec {
+    
+    private struct Dog: JSONAPIEntity {
+        let id: String
+        let dogName: String
+    }
+    
+    private struct JUser: JSONAPIEntity {
+        let id: String
+        let userName: String
+        let doggyDog: Dog
+    }
+    
+    struct User: Serializable {
+        let name: String
+    }
+    
+    struct Friend: Serializable {
+        let friends: [User]
+        
+        init(friends: [User]) {
+            self.friends = friends
+        }
+    }
+    
+    struct Snake: Serializable {
+        let theSnakeCamelFriend: String
+    }
+    
+    override func spec() {
+        
+        let user = User(name: "Alex")
+        let friend = Friend(friends: [user])
+        
+        describe("Serialization Transformers") {
+            it("transforms the keys of a Serializable objects") {
+                let serialized = UppercaseTransformer(wrapped: friend).serialize() as! [String: AnyObject]
+                let friends = serialized["FRIENDS"] as? [AnyObject]
+                let first = friends?.first as? [String: AnyObject]
+                expect(first?.keys.first).to(equal("NAME"))
+            }
+            
+            it("should apply transformations in the right order") {
+                do {
+                    let serialized = LowercaseFirstCharacterTransformer(wrapped: UppercaseTransformer(wrapped: friend)).serialize() as! [String: AnyObject]
+                    let friends = serialized["fRIENDS"] as? [AnyObject]
+                    let first = friends?.first as? [String: AnyObject]
+                    expect(first?.keys.first).to(equal("nAME"))
+                }
+                
+                do {
+                    let serialized = UppercaseTransformer(wrapped: LowercaseFirstCharacterTransformer(wrapped: friend)).serialize() as! [String: AnyObject]
+                    let friends = serialized["FRIENDS"] as? [AnyObject]
+                    let first = friends?.first as? [String: AnyObject]
+                    expect(first?.keys.first).to(equal("NAME"))
+                }
```

Comment:
Just to reuse the code and variable names 🤓 And doesn't look bad. Don't like it?